app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/package.scala

```diff
@@ -53,9 +60,43 @@ package object controllers {
 
     private val noAmpersand = Constraints.pattern("[^&]*".r, error = "error.no.ampersand")
 
+    private val agencyNameConstraint: Constraint[String] = Constraint[String] { fieldValue: String =>
+      Constraints.nonEmpty(fieldValue) match {
+        case i: Invalid => i
+        case Valid => fieldValue match {
+          case value if value.length > agencyNameMaxLength => Invalid(ValidationError("error.name.invalid"))
+          case value if !value.matches(agencyNameRegex) => Invalid(ValidationError("error.name.invalid"))
+          case _ => Valid
+        }
+      }
+    }
+    private val addressConstraint: Constraint[String] = Constraint[String] { fieldValue: String =>
+      Constraints.nonEmpty(fieldValue) match {
+        case i: Invalid => i
+        case Valid => fieldValue match{
+          case value if value.length > addressMaxLength => Invalid(ValidationError("error.addressLine1.invalid"))
+          case value if !value.matches(addressRegex) => Invalid(ValidationError("error.addressLine1.invalid"))
+          case _ => Valid
+        }
+      }
+    }
+
+    private val address23Constraint: Constraint[String] = Constraint[String] { fieldValue: String =>
+      Constraints.nonEmpty(fieldValue) match {
+        case i: Invalid => i
+        case Valid => fieldValue match{
+          case value if value.length > addressMaxLength => Invalid(ValidationError("error.addressLine.invalid"))
+          case value if !value.matches(addressRegex) => Invalid(ValidationError("error.addressLine.invalid"))
+          case _ => Valid
+        }
+      }
+    }
+
     def utr: Mapping[Utr] = nonEmptyText.transform[Utr](Utr.apply,_.value).verifying("error.utr.invalid", utr => Utr.isValid(utr.value))
     def postcode: Mapping[String] = text verifying nonEmptyPostcode
-    def telephoneNumber: Mapping[String] = text verifying nonEmptyTelephoneNumber
-    def agencyName: Mapping[String] = nonEmptyText(maxLength = 40) verifying noAmpersand
+    def telephoneNumber: Mapping[String] = text verifying TelephoneNumberConstraint
+    def agencyName: Mapping[String] = text verifying noAmpersand verifying agencyNameConstraint
```

Comment:
I think we can avoid having so many different custom constraints here.

Step 1 - this doesn't change any behaviour (I think) - remove address23Constraint because it is the same as addressConstraint (isn't it?)
```
    def addressLine1: Mapping[String] = text verifying addressConstraint
    def addressLine23: Mapping[Option[String]] = optional(text verifying addressConstraint)
```

Step 2 - this does change behaviour because the message you'd get for input too long would be different but I think this is a positive behaviour change:
```
    def agencyName: Mapping[String] = text(maxLength = 40) verifying noAmpersand verifying desPlainTextConstraint
    def addressLine1: Mapping[String] = text(maxLength = 35) verifying desPlainTextConstraint
    def addressLine23: Mapping[Option[String]] = optional(text(maxLength = 35) verifying desPlainTextConstraint)
```

desPlainTextConstraint would validate against the regex `"^[A-Za-z0-9 \\-,.&'\\/]*$"`, i.e. the same regex as address and agency name except with the length part (`{1,35}` or `{1,40}`) changed to `*`.